src/QueryM/Ogmios.purs

```diff
@@ -0,0 +1,247 @@
+-- | Provides types and instances to create Ogmios requests and decode
+-- | its responses.
+module QueryM.Ogmios where
+
+import Prelude
+
+import Aeson (class DecodeAeson, Aeson, caseAesonArray, caseAesonObject, decodeAeson, getField, getFieldOptional)
+import Control.Alt ((<|>))
+import Data.Argonaut (class EncodeJson, JsonDecodeError(TypeMismatch))
+import Data.Array (index)
+import Data.BigInt (BigInt)
+import Data.Either (Either(Left, Right), hush, note)
+import Data.Foldable (foldl)
+import Data.Map (Map)
+import Data.Map as Map
+import Data.Maybe (Maybe(Just, Nothing), fromMaybe)
+import Data.Newtype (wrap)
+import Data.String (Pattern(Pattern), indexOf, splitAt, uncons)
+import Data.Traversable (sequence)
+import Data.Tuple (uncurry)
+import Data.Tuple.Nested ((/\), type (/\))
+import Data.UInt as UInt
+import Foreign.Object (Object)
+import Foreign.Object as FO
+import QueryM.JsonWsp (JsonWspCall, JsonWspRequest, mkCallType, parseFieldToBigInt, parseFieldToString, parseFieldToUInt)
+import Serialization.Address (Slot)
+import Type.Proxy (Proxy(Proxy))
+import Types.ByteArray (ByteArray, byteArrayToHex, hexToByteArray)
+import Types.Natural (Natural)
+import Types.RedeemerTag as Tag
+import Types.Value (CurrencySymbol, TokenName, Value, mkCurrencySymbol, mkTokenName, mkValue)
+import Untagged.TypeCheck (class HasRuntimeType)
+import Untagged.Union (type (|+|), toEither1)
+
+
+-- LOCAL STATE QUERY PROTOCOL
+-- https://ogmios.dev/mini-protocols/local-state-query/
+
+-- | Queries Ogmios for the chain’s current tip.
+queryChainTipCall :: JsonWspCall Unit ChainTipQR
```

Comment:
These `Query` calls won't work as the method name is not correct (the method should be `Query`, with the actual query given in `args`